bloom-housing/bloom
# feat: feature flag and new endpoint

This PR addresses #4248 


## Description

This PR adds **feature flagging** to hide or show the side nav and creates a custom call to reduce the load time for accounts with numerous applications.

Add SHOW_LOTTERY=TRUE to the public side

One note that you will see ApplicationsIndexEnum and ApplicationsFilterEnum are very similar but based on the current implementation requiring tabs, offline convo landed on using both would be best to maintain strong typing on the BE and easy tab selection based on index + better readability on the FE.

## How Can This Be Tested/Reviewed?

This can be tested by signing into an account, applying to more than one listing, and seeing in the network tab that only one call is being made no matter how many listings there are. Additionally, you can rerun the application with the new feature flag set to false, and you will no longer see the Lottery run tab but everything else should work as expected.
## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
